src/runtime/cache.cpp

```diff
@@ -70,20 +70,16 @@ WEAK size_t full_extent(const buffer_t &buf) {
 }
 
 WEAK void copy_from_to(void *user_context, const buffer_t &from, buffer_t &to) {
-    size_t buffer_size = full_extent(from);
     halide_assert(user_context, from.elem_size == to.elem_size);
     for (int i = 0; i < 4; i++) {
         halide_assert(user_context, from.extent[i] == to.extent[i]);
         halide_assert(user_context, from.stride[i] == to.stride[i]);
     }
-    memcpy(to.host, from.host, buffer_size * from.elem_size);
+    to.host = from.host;
 }
 
 WEAK buffer_t copy_of_buffer(void *user_context, const buffer_t &buf) {
     buffer_t result = buf;
```

Comment:
Is this function a noop? It looks like it just returns a shallow copy of the buffer_t now.